test/test.js

```diff
@@ -5,16 +5,53 @@ import { test } from '../src/index.js';
 describe('test', function () {
   it('should detect as path template', function () {
     assert.isTrue(test('/{path}'));
+    // trailing slash is allowed
+    assert.isTrue(test('/{path}/'));
     assert.isTrue(test('/pets/{petId}'));
     assert.isTrue(test('/{entity}/me'));
     assert.isTrue(test('/books/{id}'));
     assert.isTrue(test('/a{test}'));
+    assert.isTrue(test('/{test}a'));
+    // parentheses are allowed
+    assert.isTrue(test('/range({x},{y})'));
+    // repeated parameter names are allowed
+    assert.isTrue(test('/range({x},{y})/secondRange({x},{y})'));
     assert.isTrue(test('/{entity}/{another-entity}/me'));
+    // NOTE: the following scenario should probably be considered invalid, query parameters are not allowed in path templates
     assert.isTrue(test('/pets?offset=0&limit=10'));
+    // -->
     assert.isTrue(test('/'));
+    // special characters in literal are allowed
+    assert.isTrue(test('/-/'));
+    assert.isTrue(test('/~/'));
+    // NOTE: the following scenario should probably be considered invalid, fragments are not allowed in path templates
+    assert.isTrue(test('/#baz'));
+    // NOTE: the following scenarios should probably be considered invalid, should we allow RFC 6570 operators in parameter names?
```

Comment:
IMHO we should **not** consider those to be invalid, I see no reason for it to introduce this constraint. They are perfectly fine as percent encoded form don't apply to them.

For template expression name following ABNF non-terminal is used: 

```abnf
 1*( unreserved / pct-encoded / sub-delims / ":" / "@" )
```

We can define `Parameter Object` with `name` fixed field of `&baz`, and it will just work seamlessly.

Reply:
The reason I was asking the question for those is because it sets expectations that parameter expansion is going to work exactly like in RFC6570, which it does not because of the modifiers on the parameter definition (outside of the template)

Reply:
I think this is the same case as https://github.com/swaggerexpert/openapi-path-templating/pull/145/files#r1867780810

Reply:
https://github.com/swaggerexpert/openapi-path-templating/pull/145/files#r1867852120 opened my mind further on this. Because we don't allow `gen-delims`, we already do introduce constraint on the `Parameter Object.name` - we require those characters to be percent encoded to be part of `Parameter Object.name`. So what you propose is to go just further...

Reply:
from the TDC today: those shouldn't fail, but the syntax "doesn't do anything" since we already have fields for the behaviour. What's important is the names of the parameters match between the template, including the special character, and the element in the description.

Reply:
ACK